Dockerfile

```diff
@@ -1,11 +1,17 @@
 FROM folioci/alpine-jre-openjdk17:latest
 
+USER root
+
 # Copy your fat jar to the container
 ENV APP_FILE mod-entity-links.jar
 # - should be a single jar file
 ARG JAR_FILE=./target/*.jar
 # - copy
 COPY ${JAR_FILE} ${JAVA_APP_DIR}/${APP_FILE}
 
+# Install latest patch versions of packages: https://pythonspeed.com/articles/security-updates-in-docker/
+RUN apk upgrade --no-cache
```

Comment:
If `apk upgrade` is moved before the COPY Docker can re-use that image layer for other images (image layer cache).

Reply:
Thanks @julianladisch, moved it before the COPY

Reply:
Sorry, I mean before the complete COPY block:

```
FROM folioci/alpine-jre-openjdk17:latest

# Install latest patch versions of packages: https://pythonspeed.com/articles/security-updates-in-docker/
USER root
RUN apk upgrade --no-cache
USER folio

# Copy your fat jar to the container
ENV APP_FILE mod-entity-links.jar
# - should be a single jar file
ARG JAR_FILE=./target/*.jar
# - copy
COPY ${JAR_FILE} ${JAVA_APP_DIR}/${APP_FILE}

# Expose this port locally in the container.
EXPOSE 8081
```

And root should be switched back to folio after the upgrade.

Reply:
Thank you again